lib/yabeda/dsl/class_methods.rb

```diff
@@ -58,24 +58,26 @@ def histogram(*args, **kwargs, &block)
       #
       # @param name [Symbol] Name of default tag
       # @param value [String] Value of default tag
-      def default_tag(name, value)
-        ::Yabeda.default_tags[name] = value
+      def default_tag(name, value, group: @group)
+        Yabeda.groups[group] ||= Yabeda::Group.new(group)
+        Yabeda.groups[group].default_tag(name, value)
       end
 
       # Redefine default tags for a limited amount of time
       # @param tags Hash{Symbol=>#to_s}
       def with_tags(**tags)
-        Thread.current[:yabeda_temporary_tags] = tags
+        previous_temp_tags = temporary_tags
+        Thread.current[:yabeda_temporary_tags] = Thread.current[:yabeda_temporary_tags].merge(tags)
         yield
       ensure
-        Thread.current[:yabeda_temporary_tags] = {}
+        Thread.current[:yabeda_temporary_tags] = previous_temp_tags
       end
 
       # Get tags set by +with_tags+
       # @api private
       # @return Hash
       def temporary_tags
-        Thread.current[:yabeda_temporary_tags] || {}
+        Thread.current[:yabeda_temporary_tags] ||= {}
```

Comment:
Should this be a concurrent hash though?

Reply:
As we're working inside single thread, I don't think that there will be any race conditions.